dbcrossbarlib/src/drivers/postgres_shared/table/create_table_sql.rs

```diff
@@ -0,0 +1,135 @@
+//! This file contains a [`rust-peg`][peg] grammar. A "PEG" is a "parser
+//! expression grammar". It's basically similar to a regular expression,
+//! except it can contain recursive rules. See the site for an overview
+//! of the basic syntax.
+//!
+//! `#quiet` and `#expected` are used in a few places to give better
+//! error messages. `#quiet` suppresses certain possible tokens (such as
+//! whitespace) from "expected ___" errors, and `#expected` allows us to
+//! insert descriptive names into those messages.
+//!
+//! [peg]: https://github.com/kevinmehall/rust-peg
+
+use super::super::{PgColumn, PgCreateTable, PgDataType, PgScalarDataType};
+use crate::schema::Srid;
+
+pub use create_table_grammar::create_table as parse;
+
+peg::parser! {
+    grammar create_table_grammar() for str {
+        /// A `CREATE TABLE` expression.
+        pub rule create_table() -> PgCreateTable
+            = ws()? i("CREATE") ws() (i("UNLOGGED") ws())? i("TABLE") ws() name:identifier() ws()? "("
```

Comment:
`CREATE [ [ GLOBAL | LOCAL ] { TEMPORARY | TEMP } | UNLOGGED ] TABLE [ IF NOT EXISTS ] table_name` 🤷 

* temporary
* temp
* if not exists

but maybe we wait to see that in the wild

Reply:
Yup. We don't pretend to try to cover the entire PostgreSQL grammar. We add things when they're needed.